web3/_utils/request.py

```diff
@@ -111,9 +110,16 @@ def get_response_from_post_request(
 def make_post_request(
     endpoint_uri: URI, data: Union[bytes, Dict[str, Any]], *args: Any, **kwargs: Any
 ) -> bytes:
-    response = get_response_from_post_request(endpoint_uri, data=data, *args, **kwargs)
-    response.raise_for_status()
-    return response.content
+    kwargs.setdefault("timeout", DEFAULT_TIMEOUT)
+    kwargs.setdefault("stream", True)
+    with get_response_from_post_request(
```

Comment:
Hi @kclowes . Looks good, but you are still missing the manual control of the `timeout`. In this case, even if you are streaming, if let's say, you set the timeout to `60 seconds`, and the RPC returns `1 byte` every `59 seconds`, the timeout will not trigger and this loop can be stuck until the RPC stops sending data

Reply:
Gotcha. I'll close this is favor of yours then. On the one hand, it seems like this is desired behavior though? Could you decrease the value of your timeout?

Reply:
Curious to get your thoughts here too @fselmo. Related to #3418

Reply:
Reducing the timeout is not helping if the server is slowly streaming a response